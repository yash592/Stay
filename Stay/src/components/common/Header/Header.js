import React from "react";
import { Text, View } from "react-native";

// reusable header sticky component with title passed down as a prop

const Header = props => {
  const { textStyle, viewStyle } = styles;
  const { headerText } = props;

  return (
    <View style={viewStyle}>
      <Text style={textStyle}>{headerText}</Text>
    </View>
  );
};

const styles = {
  viewStyle: {
    backgroundColor: "#F8F8F8",
    justifyContent: "center",
    alignItems: "center",
    height: 60,
    width: "100%",

    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    elevation: 2,
    position: "relative",
    backgroundColor: "#D0FFC8"
  },
  textStyle: {
    fontSize: 20,
    fontFamily: "sans-serif-condensed",
    color: "black"
  }
};

// Make the component available to other parts of the app
export { Header };
