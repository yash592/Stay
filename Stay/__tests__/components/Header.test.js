import "react-native";
import renderer from "react-test-renderer";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";

configure({ adapter: new Adapter() });
import { Header } from "../../src/components/common/Header";
import { shallow } from "enzyme";

describe("Header component renders correctly", () => {
  it("Shallow rendering", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
