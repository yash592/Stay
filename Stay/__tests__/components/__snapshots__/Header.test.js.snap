// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly Shallow rendering 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D0FFC8",
      "elevation": 2,
      "height": 60,
      "justifyContent": "center",
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "sans-serif-condensed",
        "fontSize": 20,
      }
    }
  />
</View>
`;
